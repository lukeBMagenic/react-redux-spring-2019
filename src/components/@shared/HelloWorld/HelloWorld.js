import React, { Component } from 'react';
import './HelloWorld.scss';

class HelloWorld extends Component {
  render() {
    return (
      <div>
        <h1>{this.props.text}</h1>
      </div>
    );
  }
}

export default HelloWorld;
