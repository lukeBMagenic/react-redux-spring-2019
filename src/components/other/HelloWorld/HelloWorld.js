import React, { Component } from 'react';
import propTypes from 'prop-types'
import './HelloWorld.scss';

class HelloWorld extends Component {

  static defaultProps = { text: 'Default Text' }; 

  render() {
    return (
      <div>
        <h1>{this.props.text}</h1>
      </div>
    );
  }
}

HelloWorld.propTypes = {
  text: propTypes.string
};

export default HelloWorld;
