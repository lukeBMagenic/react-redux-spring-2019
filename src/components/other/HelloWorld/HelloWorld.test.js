import React from 'react';
import { shallow } from "enzyme";

import HelloWorld from './HelloWorld';

describe("HelloWorld", () => {
  it("should render correctly", () => {
    const component = shallow(<HelloWorld />);

    expect(component).toMatchSnapshot();
  });
});