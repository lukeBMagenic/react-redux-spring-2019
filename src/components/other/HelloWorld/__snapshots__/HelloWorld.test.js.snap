// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld should render correctly 1`] = `
<div>
  <h1>
    Default Text
  </h1>
</div>
`;
